File: 1248-count-number-of-nice-subarrays/main.go
func numberOfSubarrays(nums []int, k int) int {
    
    hashMap := make(map[int]int)
    hashMap[0] = 1
    count, oddCount := 0, 0

    for _, num := range nums {
        if num & 1 != 0 {
            oddCount++
        }

        if value, exists := hashMap[oddCount - k]; exists {
            count += value
        }

        hashMap[oddCount]++
    }

    return count
}

File: 1248-count-number-of-nice-subarrays/main.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        
        std::unordered_map<int, int> map;
        map[0] = 1;

        int count = 0;
        int odd_count = 0;

        for (const auto& num : nums) {
            if ((num & 1) != 0) {
                odd_count++;
            }

            if (map.find(odd_count - k) != map.end()) {
                count += map[odd_count - k];
            }

            map[odd_count]++;
        }

        return count;
    }
};
